sdk/eventhub/event-hubs/src/dataTransformer.ts

```diff
@@ -49,38 +65,91 @@ export const defaultDataTransformer = {
   },
 
   /**
-   * A function that takes the body property from an AMQP message
-   * (an AMQP Data type (data section in rhea terms)) and returns the decoded message body.
-   * If it cannot decode the body then it returns the body
-   * as-is.
-   * @param body - The AMQP message body
-   * @returns decoded body or the given body as-is.
+   * A function that takes the body property from an AMQP message, which can come from either
+   * the 'data', 'value' or 'sequence' sections of an AMQP message.
+   *
+   * If the body is not a JSON string the the raw contents will be returned, along with the bodyType
+   * indicating which part of the AMQP message the body was decoded from.
+   *
+   * @param body - The AMQP message body as received from rhea.
+   * @returns The decoded/raw body and the body type.
    */
-  decode(body: unknown): any {
-    let processedBody: any = body;
+  decode(body: unknown | RheaAmqpSection): { body: unknown; bodyType: BodyTypes } {
     try {
-      if (isObjectWithProperties(body, ["content"]) && isBuffer(body.content)) {
-        // This indicates that we are getting the AMQP described type. Let us try decoding it.
-        processedBody = body.content;
-      }
-      try {
-        // Trying to stringify and JSON.parse() anything else will fail flat and we shall return
-        // the original type back
-        const bodyStr: string = processedBody.toString("utf8");
-        processedBody = JSON.parse(bodyStr);
-      } catch (err) {
-        logger.verbose(
-          "[decode] An error occurred while trying JSON.parse() on the received body. " +
-            "The error is %O",
-          err
-        );
+      if (isRheaAmqpSection(body)) {
+        switch (body.typecode) {
+          case dataSectionTypeCode:
+            return { body: tryToJsonDecode(body.content), bodyType: "data" };
+          case sequenceSectionTypeCode:
+            return { body: body.content, bodyType: "sequence" };
+          case valueSectionTypeCode:
+            return { body: body.content, bodyType: "value" };
+        }
+      } else {
+        // TODO: test case
```

Comment:
Is this a TODO because it's not done yet or is there an interesting challenge here?

Reply:
Haha, I meant to erase that but the problem with adding TODOs is you have to actually go back and grep them...